File: tuan02/bai32/bai32.py
def la_chinh_phuong(n: int) -> bool:
    if n < 0:
        return False
    left, right = 0, 10**9
    while left <= right:
        mid = (left + right) // 2
        binh_phuong = mid * mid
        if binh_phuong == n:
            return True
        elif binh_phuong < n:
            left = mid + 1
        else:
            right = mid - 1
    return False

if __name__ == "__main__":
    n = int(input())
    if la_chinh_phuong(n):
        print("YES")
    else:
        print("NO")

File: tuan02/bai32/bai32.cpp
#include <iostream>

using namespace std;

bool la_chinh_phuong(long long n) {
    if (n < 0) return false;
    long long left = 0, right = 1000000000;
    while (left <= right) {
        long long mid = left + (right - left) / 2;
        long long binh_phuong = mid * mid;
        if (binh_phuong == n) return true;
        if (binh_phuong < n) left = mid + 1;
        else right = mid - 1;
    }
    return false;
}

int main() {
    long long n;
    if (cin >> n) {
        if (la_chinh_phuong(n)) {
            cout << "YES\n";
        } else {
            cout << "NO\n";
        }
    }
    return 0;
}